internal/versions/versions.go

```diff
@@ -166,50 +166,80 @@ func (h *handler) GetReleaseImageByURL(ctx context.Context, url, pullSecret stri
 	return h.addReleaseImage(url, pullSecret)
 }
 
+// Finds a release image from the cache that matches the specified version and architecture in the following
+// priority:
+// 1. Exact CPU Architecture & OpenShift x.y.z version match
+// 2. Exact CPU Architecture & OpenShift Major Minor (x.y) version match
+// 3. Multi-arch CPU with specified CPU Architecture & OpenShift x.y.z version match
+// 4. Multi-arch CPU with specified CPU Architecture & OpenShift Major Minor (x.y) version match
 func (h *handler) getReleaseImageFromCache(openshiftVersion, cpuArchitecture string) (*models.ReleaseImage, error) {
 	if cpuArchitecture == "" {
 		// Empty implies default CPU architecture
 		cpuArchitecture = common.DefaultCPUArchitecture
 	}
+
 	// Filter Release images by specified CPU architecture.
-	releaseImages := funk.Filter(h.releaseImages, func(releaseImage *models.ReleaseImage) bool {
-		for _, arch := range releaseImage.CPUArchitectures {
-			if arch == cpuArchitecture {
-				return true
+	exactCPUArchReleaseImages := funk.Filter(h.releaseImages, func(releaseImage *models.ReleaseImage) bool {
+		return swag.StringValue(releaseImage.CPUArchitecture) == cpuArchitecture
+	})
+
+	// Filter Release images by multi-arch images containing the specified CPU architecture
+	multiArchReleaseImages := funk.Filter(h.releaseImages, func(releaseImage *models.ReleaseImage) bool {
+		if *releaseImage.CPUArchitecture == common.MultiCPUArchitecture {
+			for _, arch := range releaseImage.CPUArchitectures {
+				if arch == cpuArchitecture {
+					return true
+				}
 			}
 		}
-		return swag.StringValue(releaseImage.CPUArchitecture) == cpuArchitecture
+		return false
 	})
-	if funk.IsEmpty(releaseImages) {
+
+	if funk.IsEmpty(exactCPUArchReleaseImages) && funk.IsEmpty(multiArchReleaseImages) {
 		return nil, errors.Errorf("The requested CPU architecture (%s) isn't specified in release images list", cpuArchitecture)
 	}
+
+	majorMinorVersion, err := toMajorMinor(openshiftVersion)
```

Comment:
`majorMinorVersion` looks to be used only in `getReleaseImageByVersion` function maybe we can move this statement there?

Reply:
Modified as suggested!